confidential-data-hub/hub/src/config.rs

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2024 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{env, fs, path::Path};
+
+use anyhow::*;
+use attestation_agent::config::aa_kbc_params::AaKbcParams;
+use config::{Config, File};
+use log::{debug, info};
+use serde::Deserialize;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "ttrpc")] {
+        const DEFAULT_CDH_SOCKET_ADDR: &str = "unix:///run/confidential-containers/cdh.sock";
+    } else {
+        const DEFAULT_CDH_SOCKET_ADDR: &str = "127.0.0.1:50000";
+    }
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct KbsConfig {
+    pub name: String,
+
+    pub url: String,
+
+    pub kbs_cert: Option<String>,
+}
+
+impl KbsConfig {
+    fn new() -> Result<Self> {
+        debug!("Try to get kbc and url from env and kernel commandline.");
+        let aa_kbc_params =
+            AaKbcParams::new().context("failed to read aa_kbc_params to initialize KbsConfig")?;
+        Ok(KbsConfig {
+            name: aa_kbc_params.kbc,
+            url: aa_kbc_params.uri,
+            kbs_cert: None,
+        })
+    }
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct Credential {
+    pub resource_uri: String,
+    pub path: String,
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct ImageConfiguration {
+    /// Whether to enable image security validation
+    ///
+    /// This defaults to `false`.
+    pub security_validate: bool,
```

Comment:
I've never really liked this parameter name. This refers to signature validation, right? Why not just call it that?

We should probably add slightly more extensive comments to this struct or the sample config.

Reply:
The image policy not only enables signature verification, but also can directly `reject` or `allAll` images. How about `enable_image_policy`?

Reply:
Ignore this comment and plz see https://github.com/confidential-containers/guest-components/pull/608#discussion_r1729722755

Reply:
May be we can use `insecureAcceptAnything` like used in containers policy.json ?

Reply:
@bpradipt What do you mean by `insecureAcceptAnything`? This is a policy item that can be used in the policy.json

Reply:
I meant naming the bool as "insecure_accept_anything" or "insecureacceptanything" to indicate whether image image security validation will be done or not.

Reply:
Oh, got it. I changed the parameters and combined `security_validate` and `policy_uri` into one. Plaese help to check if it is now more straightforward